src/Controller/IndexController.php

```diff
@@ -727,11 +737,15 @@ public function model_import(Request $request, Session $session)
                     if (!$get_absents) {
                         $filter = $this->config('Absences-validation') ? 'AND `valide`>0' : null;
 
-                        // Exclude absence with remote working reason.
-                        $teleworking_reasons = $this->entityManager->getRepository(AbsenceReason::class)
-                                                               ->getRemoteWorkingDescriptions();
-                        $teleworking_exception = (!empty($teleworking_reasons) and is_array($teleworking_reasons)) ? "AND `motif` NOT IN ('" . implode("','", $teleworking_reasons) . "')" : null;
-                        $filter .= " $teleworking_exception";
+                        // Exclude absence with remote working reason only for teleworking compliants positions.
+                        $position = isset($all_positions[$elem2['poste']]) ? $all_positions[$elem2['poste']] : null;
+                        if ($position && $position['teleworking'] == 1) {
+                            $teleworking_reasons = $this->entityManager->getRepository(AbsenceReason::class)
```

Comment:
Cette requête (`$teleworking_reasons = $this->entityManager->getRepository(AbsenceReason::class)`) peut être placée en dehors de la boucle foreach $dates